iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/plugin/visitors/SpringConfigMetricsVisitor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.plugin.visitors;
+
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.yaml.visitors.YamlMetricsVisitor;
+import org.sonar.iac.springconfig.tree.api.Profile;
+import org.sonar.iac.springconfig.tree.api.SyntaxToken;
+
+import static org.sonar.iac.springconfig.plugin.SpringConfigSensor.isPropertiesFile;
+
+public class SpringConfigMetricsVisitor extends YamlMetricsVisitor {
+  public SpringConfigMetricsVisitor(FileLinesContextFactory fileLinesContextFactory, NoSonarFilter noSonarFilter) {
+    super(fileLinesContextFactory, noSonarFilter);
+  }
+
+  @Override
+  protected boolean acceptFileForLoc(InputFileContext inputFileContext) {
+    // YAML files should be processed line-by-line; properties files will be handled by a visitor
```

Comment:
1. I guess it will be handled by `YamlMetricsVisitor`? The comment is not clear enought.
2. If yes, it will not be correctly, because of `YamlMetricsVisitor:61`: `        if (!(line.isBlank() || line.startsWith("#"))) {` - in properties file comments also starts from `!`. Please adjust that and add test case.

Reply:
My idea is the following. We lose a lot of information when flattening the YAML tree, so we shouldn't directly register metrics visitor on SyntaxToken nodes for YAML files. For them, the implementation from the base class will be called to calculate metrics line-by-line. For properties files, we can use the tree. I updated the comment.
I don't quite get your question. Implementation of the YamlMetricsVisitor is not called for properties, so there shouldn't be a problem with comments. Or did you mean something else?